docs/packaging-your-sql-project/intro-to-python-scripting-and-pipelines.md

```diff
@@ -11,4 +11,143 @@ kernelspec:
+
+In general, a function should only perform one action and be named accordingly. Below are guidelines to follow when defining functions:
+
+#### Naming Conventions
+
+- Function names should be lowercase, with words separated by underscores as necessary to improve readability. Variable names follow the same convention as function names.
+- Avoid "and" in function names. Dedicate a single function to a single task.
+- Adequately describe the function’s purpose by using a verb first and a noun following it. An example of a good function name is `get_data()`, which describes the action of the function and the data it returns.
+- Same goes for parameter names in functions: banish “x” and “y” from your code and use descriptive names instead.
+
+#### Docstrings
+
+- Docstrings are used to describe the purpose of a function, specifically its parameters and what it returns. They are placed immediately after the function definition and are enclosed in `"""triple double quotes"""`.
+- Blank lines should be removed from the beginning and end of the docstring.
+- It is best to list each argument on a separate line, followed by a description of the argument.
+- See [`PEP 257` guidelines](https://peps.python.org/pep-0257/) for more information on docstrings.
+
+An example is as follows:
+
+```python
+def complex(real=0.0, imag=0.0):
+  """Form a complex number.
+
+  Keyword arguments:
+  real -- the real part (default 0.0)
+  imag -- the imaginary part (default 0.0)
+  """
+  if imag == 0.0 and real == 0.0:
+      return complex_zero
+  ...
+```
+
+#### Modularization
+
+- Modularization is the technique of breaking down your script into smaller, self-contained modules or functions. Each module or function performs a specific task, making the code more readable, maintainable, and reusable.
+- Moreover, modularization enhances collaboration as multiple developers can work on different modules simultaneously, promoting collaboration and parallel development. This speeds up the overall development process and allows teams to work on specific parts of the project independently.
+- Modularization is often used in conjunction with the main program, which calls the functions defined in the modules to execute the main logic of the script. This is discussed in more detail below.
+
+### The Main Program
+
+#### What does `if __name__ == "__main__:"` mean?
+
+Typically, a script will have a `main` function that calls other functions to execute the main logic of the script. The `main` function is called from the `if __name__=="__main__":` block, which is executed when the script is run directly (not imported as a module). This allows you to have reusable code in your script that can be imported into other scripts without unintended side effects.
+
+Inside the `if __name__=="__main__":` block, you can call the functions you defined earlier to execute the main logic of your script:
+
+```python
+if __name__=="__main__":
+    # call functions here
+    pass
+```
```

Comment:
Good place to call the `complex()` function instead of `pass`

Alternatively, explain what `pass` does

Reply:
Changed `complex()` to `divide()` because it would shadow Python's in-built `complex()` function. Also made code cell running instead of ```python and called in main (also its own running cell)